client/my-sites/site-settings/header-code/index.jsx

```diff
@@ -0,0 +1,133 @@
+import { Button, Card, Gridicon } from '@automattic/components';
+import classNames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import illustration404 from 'calypso/assets/images/illustrations/illustration-404.svg';
+import DocumentHead from 'calypso/components/data/document-head';
+import QuerySiteSettings from 'calypso/components/data/query-site-settings';
+import EmptyContent from 'calypso/components/empty-content';
+import FormTextarea from 'calypso/components/forms/form-textarea';
+import HeaderCake from 'calypso/components/header-cake';
+import InlineSupportLink from 'calypso/components/inline-support-link';
+import Main from 'calypso/components/main';
+import NavigationHeader from 'calypso/components/navigation-header';
+import { successNotice, errorNotice } from 'calypso/state/notices/actions';
+import { canCurrentUser } from 'calypso/state/selectors/can-current-user';
+import getSiteSetting from 'calypso/state/selectors/get-site-setting';
+import { saveSiteSettings } from 'calypso/state/site-settings/actions';
+import {
+	isRequestingSiteSettings,
+	isSavingSiteSettings,
+	isSiteSettingsSaveSuccessful,
+	getSiteSettingsSaveError,
+} from 'calypso/state/site-settings/selectors';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import './style.scss';
+
+const HeaderCodeSettings = () => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const [ editedCode, setEditedCode ] = useState( '' );
+
+	const siteId = useSelector( ( state ) => getSelectedSiteId( state ) );
+	const headCode = useSelector( ( state ) =>
+		getSiteSetting( state, siteId, 'jetpack_header_code' )
+	);
+	const isAdmin = useSelector( ( state ) => canCurrentUser( state, siteId, 'manage_options' ) );
+	const isRequestingSettings = useSelector( ( state ) =>
+		isRequestingSiteSettings( state, siteId )
+	);
+	const isSaveFailure = useSelector( ( state ) => getSiteSettingsSaveError( state, siteId ) );
+	const isSaveSuccess = useSelector( ( state ) => isSiteSettingsSaveSuccessful( state, siteId ) );
+	const isSavingSettings = useSelector( ( state ) => isSavingSiteSettings( state, siteId ) );
+	const siteSlug = useSelector( ( state ) => getSelectedSiteSlug( state, siteId ) );
+	const saveSettings = () => {
+		dispatch(
+			saveSiteSettings( siteId, {
+				jetpack_header_code: editedCode,
+			} )
+		);
+	};
+
+	useEffect( () => {
+		const noticeOptions = {
+			duration: 4000,
+		};
+
+		if ( isSaveSuccess ) {
+			dispatch( successNotice( translate( 'Code updated successfully!' ), noticeOptions ) );
```

Comment:
I'd prefer a dot, no need for an exclamation mark here.

```suggestion
			dispatch( successNotice( translate( 'Code updated successfully.' ), noticeOptions ) );
```